test/qa/test_combinations.py

```diff
@@ -8,102 +8,31 @@
-    network,
 )
+from test_connect import disconnect_base_test
+from test_connect6 import connect_base_test
 
 
-def setup_module(module):  # noqa: ARG001
+def setup_function(function):  # noqa: ARG001
     daemon.start()
     login.login_as("default")
-
-
-def teardown_module(module):  # noqa: ARG001
-    sh.nordvpn.logout("--persist-token")
-    daemon.stop()
-
-
-def setup_function(function):  # noqa: ARG001
     logging.log()
 
 
 def teardown_function(function):  # noqa: ARG001
     logging.log(data=info.collect())
     logging.log()
 
-
-@pytest.mark.parametrize("group", lib.STANDARD_GROUPS)
-@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.TECHNOLOGIES)
-@pytest.mark.flaky(reruns=2, reruns_delay=90)
-@timeout_decorator.timeout(40)
-def test_connect_to_standard_group(tech, proto, obfuscated, group):
-    lib.set_technology_and_protocol(tech, proto, obfuscated)
-
-    output = sh.nordvpn.connect(group, _tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
-
-
-@pytest.mark.parametrize("group", lib.ADDITIONAL_GROUPS)
-@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.STANDARD_TECHNOLOGIES)
-@pytest.mark.flaky(reruns=2, reruns_delay=90)
-@timeout_decorator.timeout(40)
-def test_connect_to_additional_group(tech, proto, obfuscated, group):
-    lib.set_technology_and_protocol(tech, proto, obfuscated)
-
-    output = sh.nordvpn.connect(group, _tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
-
-
-@pytest.mark.parametrize(("target_tech", "target_proto", "target_obfuscated"), lib.STANDARD_TECHNOLOGIES)
-@pytest.mark.parametrize(("source_tech", "source_proto", "source_obfuscated"), lib.TECHNOLOGIES)
-@pytest.mark.flaky(reruns=2, reruns_delay=90)
-@timeout_decorator.timeout(40)
-def test_reconnect_matrix_standard(
-        source_tech,
-        target_tech,
-        source_proto,
-        target_proto,
-        source_obfuscated,
-        target_obfuscated,
-):
-    lib.set_technology_and_protocol(source_tech, source_proto, source_obfuscated)
-
-    output = sh.nordvpn.connect(_tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    lib.set_technology_and_protocol(target_tech, target_proto, target_obfuscated)
-
-    output = sh.nordvpn.connect(_tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
+    sh.nordvpn.logout("--persist-token")
```

Comment:
shouldn't this be wrapped in some try/catch in case one of the commands fails, so the test is not marked as failed?

Reply:
Answered here:
https://github.com/NordSecurity/nordvpn-linux/pull/463#discussion_r1647108671